test/SILOptimizer/assemblyvision_remark/cast_remarks.swift

```diff
@@ -73,14 +70,11 @@ public func condCast3<NS, T>(_ ns: NS) -> T? {
 
 public func condCast4<NS, T>(_ ns: NS, _ ns2: NS) -> T? {
   // Make sure the colon info is right so that the arrow is under the a.
-  //
-  // Today, we lose that x was assigned ns2. This is flow sensitive information
-  // that we might be able to recover. We still emit that a runtime cast
-  // occurred here, just don't say what the underlying value was.
   var x = ns
   x = ns2
   return x as? T  // expected-remark @:12 {{conditional runtime cast of value with type 'NS' to 'T'}}
-                  // expected-note @-9:42 {{of 'ns2'}}
+                  // expected-note @-5:42 {{of 'ns2'}}
+                  // expected-note @-4:7 {{of 'x'}}
```

Comment:
Nice, so this actually fixes another bug?

Reply:
Yes, now that getDecl() returns the correct variable declaration after a store, this pass is able to point back to the `x` value! After optimizations. the SILValue will be associated with both debug variables, so the pass prints both of them.